lib/rom/factory/sequences.rb

```diff
@@ -19,17 +19,20 @@ def self.[](relation)
 
       # @api private
       def initialize
+        @mutex = Mutex.new
         reset
       end
 
       # @api private
       def next(key)
-        registry[key] += 1
+        @mutex.synchronize do
+          registry[key] += 1
+        end
```

Comment:
concurrent-ruby has `AtomicFixnum` which suits better. However, it's an open question if running a single spec in threaded env with a sequence from rom-factory makes sense at all :)

Reply:
It's a singleton, so my gut feeling is that it was made that way to make it concurrent safe for potential in process spec parallelization. I did not want to introduce any new types and wanted to stick with the current API.